src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -0,0 +1,557 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace System.IO
+{
+    /// <devdoc>
+    ///    Listens to the system directory change notifications and
+    ///    raises events when a directory or file within a directory changes.
+    /// </devdoc>
+
+    public partial class FileSystemWatcher : IDisposable
+    {
+        /// <devdoc>
+        ///     Private instance variables
+        /// </devdoc>
+        // Directory being monitored
+        private string _directory;
+
+        // Filter for name matching
+        private string _filter;
+
+        // The watch filter for the API call.
+        private const NotifyFilters c_defaultNotifyFilters = NotifyFilters.LastWrite | NotifyFilters.FileName | NotifyFilters.DirectoryName;
+        private NotifyFilters _notifyFilters = c_defaultNotifyFilters;
+
+        // Flag to watch subtree of this directory
+        private bool _includeSubdirectories = false;
+
+        // Flag to note whether we are attached to the thread pool and responding to changes
+        private bool _enabled = false;
+
+        // Buffer size
+        private int _internalBufferSize = 8192;
+
+        // Used for synchronization
+        private bool _disposed;
+
+        // Event handlers
+        private FileSystemEventHandler _onChangedHandler = null;
+        private FileSystemEventHandler _onCreatedHandler = null;
+        private FileSystemEventHandler _onDeletedHandler = null;
+        private RenamedEventHandler _onRenamedHandler = null;
+        private ErrorEventHandler _onErrorHandler = null;
+
+        // To validate the input for "path"
+        private static readonly char[] s_wildcards = new char[] { '?', '*' };
+
+        private const int c_notifyFiltersValidMask = (int)(NotifyFilters.Attributes |
+                                                           NotifyFilters.CreationTime |
+                                                           NotifyFilters.DirectoryName |
+                                                           NotifyFilters.FileName |
+                                                           NotifyFilters.LastAccess |
+                                                           NotifyFilters.LastWrite |
+                                                           NotifyFilters.Security |
+                                                           NotifyFilters.Size);
+
+        static FileSystemWatcher()
+        {
+#if DEBUG
```

Comment:
Since all of the static ctor's contents are under DEBUG, the cctor itself should probably be as well.